src/plugins/expressions/common/execution/execution.ts

```diff
@@ -44,6 +45,18 @@ import { ExecutionContract } from './execution_contract';
 import { ExpressionExecutionParams } from '../service';
 import { createDefaultInspectorAdapters } from '../util/create_default_inspector_adapters';
 
+type UnwrapReturnType<Function extends (...args: any[]) => unknown> =
```

Comment:
can you explain this please ?

Reply:
That's a macro to extract the return type of an expression function. The expression function can either return an observable, or a promise, or a naked value. We gather the return value type here no matter the way it is returned.